env0/resource_environment.go

```diff
@@ -876,17 +885,68 @@ func resourceEnvironmentDelete(ctx context.Context, d *schema.ResourceData, meta
 		return diag.Errorf(`must enable "force_destroy" safeguard in order to destroy`)
 	}
 
-	_, err := apiClient.EnvironmentDestroy(d.Id())
+	res, err := apiClient.EnvironmentDestroy(d.Id())
 	if err != nil {
 		if frerr, ok := err.(*http.FailedResponseError); ok && frerr.BadRequest() {
 			tflog.Warn(ctx, "Could not delete environment. Already deleted?", map[string]interface{}{"id": d.Id(), "error": frerr.Error()})
 			return nil
 		}
 		return diag.Errorf("could not delete environment: %v", err)
 	}
+
+	if d.Get("wait_for_destroy").(bool) {
+		if err := waitForEnvironmentDestroy(apiClient, res.Id); err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
 	return nil
 }
 
+func waitForEnvironmentDestroy(apiClient client.ApiClientInterface, deploymentId string) error {
+	waitInteval := time.Second * 10
+	timeout := time.Minute * 30
+
+	if os.Getenv("TF_ACC") == "1" { // For acceptance tests reducing interval to 1 second and timeout to 10 seconds.
+		waitInteval = time.Second
+		timeout = time.Second * 10
+	}
+
+	ticker := time.NewTicker(waitInteval) // When invoked - check the status.
+	timer := time.NewTimer(timeout)       // When invoked - timeout.
+	results := make(chan error)
+
+	go func() {
+		for {
+			deployment, err := apiClient.EnvironmentDeployment(deploymentId)
+			if err != nil {
+				results <- fmt.Errorf("failed to get environment deployment '%s': %w", deploymentId, err)
+				return
+			}
+
+			if slices.Contains([]string{"WAITING_FOR_USER", "TIMEOUT", "FAILURE", "CANCELLED", "INTERNAL_FAILURE", "ABORTING", "ABORTED", "SKIPPED", "NEVER_DEPLOYED"}, deployment.Status) {
```

Comment:
I suggest to not fail on `WAITING_FOR_USER` but rather keep on waiting, and maybe write something back to the user that the deployment is waiting for an approval

This is actually something to consider for other statuses as well, like `QUEUED`, so how about we print the current status on every iteration?

Reply:
Added additional messages, and no longer fast failing with WAITING_FOR_USER